Markdown task

Andreyinthesky opened the pull request.

Commit:
Init Markdown project

Commit:
Markdown task complete
+ Add some tests in Md_Should.cs

Commit:
Refactoring
+ StringExtensions tests
+ MarkdownTokenizer tests
+ test moved to own folder

lurlrlrl commented on cs/Markdown/MarkdownTests/Md_Should.cs:
```diff
@@ -0,0 +1,61 @@
+        {
+            md = new Md();
+        }
+
```
В спецификации явно сказано: "Любой символ можно экранировать, чтобы он не считался частью разметки."
\ - тоже символ и его можно экранировать.

lurlrlrl replied:
Вот тесты на граничные случаи, как минимум они должны проходить:
```
[TestCase(@"\\", ExpectedResult = @"\")]
[TestCase(@"\\_a_", ExpectedResult = @"\<em>a</em>")]
```

lurlrlrl commented on cs/Markdown/md.cs:
```diff
@@ -0,0 +1,51 @@
+        public string ConvertToHtmlString(string paragraph, List<TagType> tagTypes)
+        {
+            var result = new StringBuilder();
+            var markdownTokenizer = new MarkdownTokenizer(paragraph, tagTypes);
```
Рендерер создаёт токенайзер, при этом с разными настройками - это явно не его ответственность.

Commit:
Refactoring Md
+ Add Md test cases, add performance test cases
+ Update StringExtension + tests
+ Refactroring TagType and child classes

Commit:
Add <del> tag support
+ Refactoring
+ Add more tests in Md and Md_Tokenizer
+ Update specification

Andreyinthesky commented:
> Рендерер создаёт токенайзер, при этом с разными настройками - это явно не его ответственность.

Я не смог решить эту проблему, если я не смогу в процессе настраивать токенайзер, то я не могу исключить тэги которых на данный момент в строке быть не может (например, strong не может содержать del, но передать это условие не получается)

Commit:
Fix escaped tags
+ add new tests
